"panic: send on closed channel" after call Close on peerConnection

### Your environment.
  - Version: github.com/pion/mediadevices v0.3.2, github.com/pion/webrtc/v3 v3.1.10
  - Browser: Chrome 98.0.4
  - Other Information:
####  Stacktrace:

```
panic: send on closed channel

goroutine 369 [running]:
github.com/pion/mediadevices.(*baseTrack).unbind(0x3962393261333461, {{0xc00027c8d0, 0x20}, {{0x7ff693bfa658, 0x0, 0x0}, {0xc000f00000, 0x1, 0x1}}, 0xec9d1d4, ...})
	$GOPATH/pkg/mod/github.com/pion/mediadevices@v0.3.2/track.go:227 +0x70
github.com/pion/mediadevices.(*VideoTrack).Unbind(0x0, {{0xc00027c8d0, 0x20}, {{0x7ff693bfa658, 0x0, 0x0}, {0xc000f00000, 0x1, 0x1}}, 0xec9d1d4, ...})
	$GOPATH/pkg/mod/github.com/pion/mediadevices@v0.3.2/track.go:311 +0x45
github.com/pion/webrtc/v3.(*RTPSender).ReplaceTrack(0xc00016c2c0, {0x0, 0x0})
	$GOPATH/pkg/mod/github.com/pion/webrtc/v3@v3.1.10/rtpsender.go:157 +0x198
github.com/pion/webrtc/v3.(*RTPSender).Stop(0xc00016c2c0)
	$GOPATH/pkg/mod/github.com/pion/webrtc/v3@v3.1.10/rtpsender.go:240 +0x85
github.com/pion/webrtc/v3.(*RTPTransceiver).Stop(0xc00019a510)
	$GOPATH/pkg/mod/github.com/pion/webrtc/v3@v3.1.10/rtptransceiver.go:148 +0x7e
github.com/pion/webrtc/v3.(*PeerConnection).Close(0xc000029200)
	$GOPATH/pkg/mod/github.com/pion/webrtc/v3@v3.1.10/peerconnection.go:1878 +0x1c5
gitlab.com/awati/vrp-server/peerconnection.CloseConnection(0xc0000a3fd0, 0x23c)
	$PROJECT_FOLDER/peerconnection/peerconnection.go:360 +0x148
gitlab.com/awati/vrp-server/peerconnection.StartEngine.func1(0x5)
	$PROJECT_FOLDER/peerconnection/peerconnection.go:117 +0x48
created by github.com/pion/webrtc/v3.(*PeerConnection).onConnectionStateChange
	$GOPATH/pkg/mod/github.com/pion/webrtc/v3@v3.1.10/peerconnection.go:488 +0x175
```

### What did you do?
I call a close function when the state changes to some imcompatible status:
```go
 peerConnection.OnConnectionStateChange(func(pcs webrtc.PeerConnectionState) {
 	switch pcs {
	case webrtc.PeerConnectionStateDisconnected, webrtc.PeerConnectionStateClosed, webrtc.PeerConnectionStateFailed:
 		CloseConnection(peerConnection, pid)
 		fallthrough
 		...
```
and inside the CloseConnection i call peerConnection.Close()


### What did you expect?
To close the connection and continue to run the program
 
### What happened?
the process end with a panic message

--- a/pkg/avfoundation/avfoundation_darwin.go
+++ b/pkg/avfoundation/avfoundation_darwin.go
@@ -118,8 +118,8 @@ func (rc *ReadCloser) dataCb(data []byte) {
 		return
 	}
 	select {
+	// Use the Done channel to avoid waiting for new data from closed camera
 	case <-rc.cancelCtx.Done():
-		close(rc.dataChan)
 	case rc.dataChan <- data:
 	}
 }
@@ -141,6 +141,7 @@ func (rc *ReadCloser) Close() {
 		rc.onClose()
 	}
 	rc.cancelFunc()
+	close(rc.dataChan)
 	unregister(rc.id)
 }
 

--- a/pkg/avfoundation/avfoundation_darwin.go
+++ b/pkg/avfoundation/avfoundation_darwin.go
@@ -147,9 +147,9 @@ func (rc *ReadCloser) Close() {
 		rc.onClose()
 	}
 	rc.cancelFunc()
+	unregister(rc.id)
 	rc.closeWG.Wait()
 	close(rc.dataChan)
-	unregister(rc.id)
 }
 
 // Session represents a capturing session.

--- a/pkg/avfoundation/avfoundation_darwin.go
+++ b/pkg/avfoundation/avfoundation_darwin.go
@@ -14,6 +14,7 @@ import (
 	"context"
 	"fmt"
 	"io"
+	"sync"
 	"unsafe"
 
 	"github.com/pion/mediadevices/pkg/frame"
@@ -99,6 +100,7 @@ type ReadCloser struct {
 	onClose    func()
 	cancelCtx  context.Context
 	cancelFunc func()
+	closeWG    *sync.WaitGroup
 }
 
 func newReadCloser(onClose func()) *ReadCloser {
@@ -109,10 +111,14 @@ func newReadCloser(onClose func()) *ReadCloser {
 	cancelCtx, cancelFunc := context.WithCancel(context.Background())
 	rc.cancelCtx = cancelCtx
 	rc.cancelFunc = cancelFunc
+	rc.closeWG = &sync.WaitGroup{}
 	return &rc
 }
 
 func (rc *ReadCloser) dataCb(data []byte) {
+	rc.closeWG.Add(1)
+	defer rc.closeWG.Done()
+
 	// TODO: add a policy for slow reader
 	if rc.cancelCtx.Err() != nil {
 		return
@@ -141,6 +147,7 @@ func (rc *ReadCloser) Close() {
 		rc.onClose()
 	}
 	rc.cancelFunc()
+	rc.closeWG.Wait()
 	close(rc.dataChan)
 	unregister(rc.id)
 }

--- a/track.go
+++ b/track.go
@@ -188,7 +188,8 @@ func (track *baseTrack) bind(ctx webrtc.TrackLocalContext, specializedTrack Trac
 			close(stopRead)
 			encodedReader.Close()
 
-			// When there's another call to unbind, it won't block since we mark the signalCh to be closed
+			// When there's another call to unbind, it won't block since we remove the current ctx from active connections
+			track.removeActivePeerConnection(ctx.ID())
 			close(signalCh)
 			if doneCh != nil {
 				close(doneCh)
@@ -256,9 +257,10 @@ func (track *baseTrack) bind(ctx webrtc.TrackLocalContext, specializedTrack Trac
 }
 
 func (track *baseTrack) unbind(ctx webrtc.TrackLocalContext) error {
-	ch, err := track.removeActivePeerConnection(ctx.ID())
-	if err != err {
-		return err
+	ch := track.removeActivePeerConnection(ctx.ID())
+	// If there isn't a registered chanel for this ctx, it means it has already been unbound
+	if ch == nil {
+		return nil
 	}
 
 	doneCh := make(chan struct{})
@@ -267,17 +269,17 @@ func (track *baseTrack) unbind(ctx webrtc.TrackLocalContext) error {
 	return nil
 }
 
-func (track *baseTrack) removeActivePeerConnection(id string) (chan<- chan<- struct{}, error) {
+func (track *baseTrack) removeActivePeerConnection(id string) chan<- chan<- struct{} {
 	track.mu.Lock()
 	defer track.mu.Unlock()
 
 	ch, ok := track.activePeerConnections[id]
 	if !ok {
-		return nil, errNotFoundPeerConnection
+		return nil
 	}
 	delete(track.activePeerConnections, id)
 
-	return ch, nil
+	return ch
 }
 
 func newTrackFromDriver(d driver.Driver, constraints MediaTrackConstraints, selector *CodecSelector) (Track, error) {

